### ISSUE TITLE
[Feature]: OTLP Exporter Retry and Persistence

### ISSUE THREAD

**Issue by AaronRM** [open-telemetry/opentelemetry-rust#2364]
### Describe the solution you'd like:

Per [the spec for OTLP Exporters](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#retry), "Transient errors MUST be handled with a retry strategy. This retry strategy MUST implement an exponential back-off with jitter to avoid overwhelming the destination until the network is restored or the destination has recovered."

Similar to [the equivalent feature for opentelemetry-dotnet](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4115), the Rust OpenTelemetry implementation must support retry of failed requests to be conformant to the OTLP Exporter spec. The retry mechanism should also be able to leverage an optional persistent store for additional resilience to unexpected process restarts. Implementation should follow a similar pattern to the dotnet implementation.

One set of PRs should implement retry logic for the OTLP Exporter per the OpenTelemetry spec.

Another set of PRs should introduce persistent blob abstractions (i.e. similar to [OpenTelemetry.PersistentStorage.Abstractions](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.PersistentStorage.Abstractions) and a file-based implementation (i.e. similar to [OpenTelemetry.PersistentStorage.FileSystem](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.PersistentStorage.FileSystem)).  This set of PRs should also include updates to the retry logic to allow opt-in use persistent storage for retry.

# References
- [OTLP Specification](https://opentelemetry.io/docs/specs/otlp/)
- [OTLP Exporter Specification](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#retry)
- [[opentelemetry-dotnet] OTLP Exporter Retry and Persist Feature](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1278)
- [[opentelemetry-dotnet] Add persistent storage to exporter](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4115)

**Comment by cijothomas** [comment]
Correct. Persistence storage is not a blocker for stable release. Its a nice-to-have capability, and is not spec-ed yet.

**Comment by scottgerring** [comment]
Great; marking as blocked on spec.

**Comment by scottgerring** [comment]
Removing from the otlp stabilisation milestone as we've agreed this doesn't need to be done before then

**Comment by scottgerring** [comment]
Hey @TommyCpp , i've picking through the OTLP backlog working out what we still need. I'd like to close this in favour of some more specific issues and PRs, and acknowledge completed work:

### Done
   - **Endpoint (HTTP + gRPC)** — implemented, including signal-specific variants and auto-appended signal paths
   - **Headers** — implemented via `OTEL_EXPORTER_OTLP_HEADERS` + signal-specific variants
   - **Compression** — implemented via `OTEL_EXPORTER_OTLP_COMPRESSION` (gzip + zstd)
 
### Outstanding

   - **Insecure** — `OTEL_EXPORTER_OTLP_INSECURE` tracked in PR #3365
   - **Certificate file / Client key file / Client certificate file** — `OTEL_EXPORTER_OTLP_CERTIFICATE`, `OTEL_EXPORTER_OTLP_CLIENT_KEY`, `OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE` tracked in PR #3364 and PR #3047
   - **Timeout** — env var is read; enforcement gaps tracked in #2576
   - **Protocol** — `OTEL_EXPORTER_OTLP_PROTOCOL` is read; `FromStr` parsing fix tracked in PR #2758
